lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/stack/StackPresenter.java

```diff
@@ -244,24 +244,38 @@ private void applyTopBarOptions(Options options, StackController stack, ViewCont
 
     private void applyStatusBarDrawBehindOptions(TopBarOptions topBarOptions, Options withDefault) {
         if (withDefault.statusBar.visible.isTrueOrUndefined() && withDefault.statusBar.drawBehind.isTrue()) {
-            topBar.setTopPadding(StatusBarUtils.getStatusBarHeight(activity));
-            topBar.setHeight(topBarOptions.height.get(UiUtils.getTopBarHeightDp(activity)) + StatusBarUtils.getStatusBarHeightDp(activity));
+            topBar.setTopPadding(SystemUiUtils.getStatusBarHeight(activity));
+            topBar.setHeight(topBarOptions.height.get(UiUtils.getTopBarHeightDp(activity)) + SystemUiUtils.getStatusBarHeightDp(activity));
+
         } else {
             topBar.setTopPadding(0);
             topBar.setHeight(topBarOptions.height.get(UiUtils.getTopBarHeightDp(activity)));
         }
     }
-
-    private void mergeStatusBarDrawBehindOptions(TopBarOptions topBarOptions, Options toMerge) {
-        if (toMerge.statusBar.drawBehind.hasValue()) {
-            if (toMerge.statusBar.visible.isTrueOrUndefined() && toMerge.statusBar.drawBehind.isTrue()) {
-                topBar.setTopPadding(StatusBarUtils.getStatusBarHeight(activity));
-                topBar.setHeight(topBarOptions.height.get(UiUtils.getTopBarHeightDp(activity)) + StatusBarUtils.getStatusBarHeightDp(activity));
-            } else {
-                topBar.setTopPadding(0);
-                topBar.setHeight(topBarOptions.height.get(UiUtils.getTopBarHeightDp(activity)));
+    
+    private void mergeStatusBarDrawBehindOptions(TopBarOptions topBarOptions, Options childOptions) {
```

Comment:
look like this code does the same thing, maybe we shouldn't touch it?

Reply:
it has missing use cases, when applying options (initial top bar position) the code is alright, but when changing the position we need to take into consideration more cases, see the differences, each toggle is a `mergeOptions` with a single option change:

old code:


https://user-images.githubusercontent.com/7227793/143925712-fd70a1a4-cc82-4e40-abe2-a141d8b20b35.mov

new code:


https://user-images.githubusercontent.com/7227793/143925808-12c62c32-dd4b-4420-9934-04576bd30b1e.mov